File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        res = strs[0]
        for s in strs:
            res = res[:len(s)]
            for i in range(len(s)):
                if i==len(res) or s[i]!=res[i]:
                    res = res[:i]
                    break
                    
        return res

File: 14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = strs[0];
        for(int i=1;i<strs.size();i++){
            for(int j=0;j<ans.size();j++){
                // cout<<ans[j]<<","<<strs[i][j]<<endl;
                if(ans[j]!=strs[i][j]){
                    ans.erase(ans.begin()+j,ans.end());
                    break;
                }
            }
        }
        return ans;
    }
};